addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,272 @@
+        }
+
+        /**
+         * Check if the element is in the viewport.
+         * @private
+         * @static
+         * @param {HTMLElement}
+         * @returns {Boolean} True if the element currently fits inside 
+         *                    the viewport, false otherwise.
+         */
+        static _isInViewport(element) {
+            const rect = element.getBoundingClientRect();
+            const html = document.documentElement;
+            return (
+                rect.top >= 0 &&
+                rect.left >= 0 &&
+                rect.bottom <= (window.innerHeight || html.clientHeight) &&
+                rect.right <= (window.innerWidth || html.clientWidth)
+            );
+        }
+
+        /**
+         * Hide any displayed popover, display the popover in argument and mark
+         * it as the one being displayed in the class attribute. Add an event
+         * listener on the document to detect any click outside the popover as a
+         * closing event.
+         * Note that the event listener is added for the capture phase, meaning
+         * that it will *always* run before the click listener set on a popover
+         * template. This means that there can be no race condition regarding
+         * clicking on another popover trigger: any existing popover will be
+         * removed during the capture phase, then a new listener will be added
+         * in the bubbling phase of the new popover' trigger click event.
+         * @static
+         * @param {Popover} popover The popover component to display.
+         */
+        static display(popover) {
+            /* this should never happen because of the way the events are handled,
+            but in case an imaginative dev puts a popover inside another one, i'd
+            prefer the first one to be hidden anyway, causing the second one to
+            never display itself (since its target won't be in the viewport) */
+            if (this.displayed) {
+                this.displayed._close();
+            }
+            if (!this.isListening) {
+                document.addEventListener('click', Popover.documentClickHandler, { capture: true });
+                this.isListening = true;
+            }
+            this.displayed = popover;
+            popover.state.displayed = true;
+        }
+
+        /**
+         * Hide any displayed popover and remove event listeners if no popover is
+         * about to replace the current one.
+         * @static
+         * @param {Boolean} nextPopover: whether a new popover is about to be opened; if true,
+         * the document click listener will not be removed.
+         */
+        static hide(nextPopover) {
+            const popover = this.displayed;
+            // only remove the listener of we are not about to open a new popover
+            if (this.isListening && !nextPopover) {
+                document.removeEventListener('click', Popover.documentClickHandler, { capture: true });
```

Comment:
why not this.constructor.documentClickHandler here (and where you bind it, of course)?

Reply:
but since we can make it work without binding i'll probably just use the function as it is